### PR TITLE
Enable specifying custom response headers

### PR DESCRIPTION
**Pull request by ellsclytn**
This allows us to respond with more than just HTML. Use case: BRB application which is mostly a React app, but also needs to also function as a basic JSON API.

### CODE REVIEW

File: lib/prod-server/app.js [hunk 1]
```diff
@@ -53,15 +53,21 @@ module.exports = ({
   }
   server.use((req, res) => {
     const url = req.url.replace(baseMatch, '/')
-    runner(url).then(({ result, redirect }) => {
+    runner(url).then(({ result, redirect, options = {} }) => {
```
Comment (bradparker) [inline lib/prod-server/app.js]: I don't like `options` 

`{ result, redirect, status, headers, body }` would be better I think.

If you want the "more complete" behaviour you can use the three new keys (`status, headers, body`).

File: lib/prod-server/app.js [hunk 2]
```diff
@@ -53,15 +53,21 @@ module.exports = ({
...
       if (redirect) {
         res.statusCode = 301
         res.setHeader('Location', redirect)
         res.setHeader('Cache-Control', 'no-cache')
         res.end()
       } else {
         res.setHeader('Content-Type', 'text/html')
-        res.statusCode = 200
+
+        const { headers = [] } = options
```
Comment (bradparker) [inline lib/prod-server/app.js]: You can have this as an object and iterate using `entries`.

```
const headers = { 'Content-type': 'application/json' }
entries(headers).forEach(([key, value]) => {
  // stuff
})
```

Making a for a slightly nicer API.

I _think_ lodash even supplies a forEach for objects which cuts out the entries middleman.

Comment (ellsclytn) [reply lib/prod-server/app.js]: Oo, really good idea.

File: lib/prod-server/app.js (hunk 1 above)
Comment (ellsclytn) [reply lib/prod-server/app.js]: How would `body` differ from `result`?

Comment (bradparker) [reply lib/prod-server/app.js]: It wouldn't at all, except that it's more in line with an object that represents a HTTP response.

**Pull request merged**